File: Perl/bench.pl
use strict;
use v5.18;
use Time::HiRes qw(time);

my $time_begin = time;

my $num_rows = 100000;
my $num_cols = 10;
my $count = 0;
my $xx = "x" x 1000;
    
printf("%d MB\n", `ps -o rss= -p $$` / 1024);

sub gen {
    $count++;
    return $count . $xx;
}

sub makeCSV {
    my @data = map { [ map { gen } (1..$num_cols) ] } (1..$num_rows);
    say $count;
    my $csv = join("\n", map { join(",", @$_) } @data);
#say $csv;
#end
    printf("%d MB\n", `ps -o rss= -p $$` / 1024);
    return $csv;
}

for my $i (1..10) {
#    say "iteration " . $i;
    say length(makeCSV);
}
    
printf("%.2fs\n", time - $time_begin);

File: Ruby/bench.rb
require "benchmark"

#GC.disable

time = Benchmark.realtime do
    
    $num_rows = 100000
    $num_cols = 10
    $count = 0
    $xx = "x"*1000

    puts "%d MB" % (`ps -o rss= -p #{Process.pid}`.to_i/1024)
    
    def gen
        $count+=1
        return $count.to_s + $xx
    end

    def makeCSV
        data = Array.new($num_rows) { Array.new($num_cols) { gen } }
        puts $count
        csv = data.map { |row| row.join(",") }.join("\n")
        puts "%d MB" % (`ps -o rss= -p #{Process.pid}`.to_i/1024)
        return csv
    end
    
    for i in 1..10
#    say "iteration " . $i;
        puts makeCSV.length
    end

end

puts time.round(2)
